integration-tests/bats/checkout.bats

```diff
@@ -178,3 +178,42 @@ SQL
   cmd=$(echo "${lines[1]}" | cut -d ' ' -f 1,2,3)
   [[ $cmd =~ "dolt checkout $sha" ]]
 }
+
+@test "commit --amend: only commit message is changed" {
+  dolt sql -q "create table test (id int primary key);"
+  dolt sql -q 'insert into test (id) values (8);'
+  dolt add .
+  dolt commit -m "original commit message"
+
+  dolt commit --amend -m "modified_commit_message"
+
+  commitmsg=$(dolt log --oneline | head -n 1)
+  [[ $commitmsg =~ "modified_commit_message" ]]
+
+  numcommits=$(dolt log --oneline | wc -l)
+  [[ $numcommits =~ "2" ]]
+
+  run dolt sql -q 'select * from test;'
+  [[ "$output" =~ "8" ]]
+}
+
+@test "commit --amend adds new changes to existing commit" {
+  dolt sql -q "create table test (id int primary key);"
+  dolt sql -q 'insert into test (id) values (8);'
+  dolt add .
+  dolt commit -m "original commit message"
+
+  dolt sql -q 'insert into test (id) values (9);'
+  dolt add .
+  dolt commit --amend -m "modified_commit_message"
+
+  commitmsg=$(dolt log --oneline | head -n 1)
+  [[ $commitmsg =~ "modified_commit_message" ]]
+
+  numcommits=$(dolt log --oneline | wc -l)
+  [[ $numcommits =~ "2" ]]
+
+  run dolt sql -q 'select * from test;'
```

Comment:
A little BATS trivia... 

using `run` won't error out if the command fails, so it's good to assert the status after using run, like this:
```
    [ "$status" -eq 0 ]
```

For the output matching, add a ``` || false ``` at the end of each output matching line, to make sure bash will error out on the right line. This one has bitten us before where not including that makes the failing output harder to debug. ex: 
```
    [[ "$output" =~ "8" ]] || false
```

Reply:
I see, thanks for clearing this up!